runtime-components/non-repudiation/src/main/scala/com/daml/nonrepudiation/SignedPayload.scala

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.nonrepudiation
+
+object SignedPayload {
+
+  def apply(
+      algorithm: AlgorithmString,
+      fingerprint: FingerprintBytes,
+      payload: PayloadBytes,
+      signature: SignatureBytes,
+  ): SignedPayload =
+    new SignedPayload(algorithm, fingerprint, payload, signature)
+
+}
+
+// Purposefully not a case class because it contains mutable arrays
```

Comment:
Do we actually need to carry mutable arrays through everything or could we wrap it in `ArraySeq.unsafeWrapArray` once at the library border and then stop worrying about it?

Reply:
That's a very good point, thanks!

Reply:
Addressed by https://github.com/digital-asset/daml/pull/8709/commits/a5e1b941f3408e9456d6d6a183b382ada2b2cae3